### PR TITLE
API: replace SHAP vector decomposition with SHAP vector projection

### PR DESCRIPTION
**Pull request by j-ittner**
This PR implements a simpler and more robust approach to calculate synergy and redundancy, based on projecting synergy and redundancy vectors onto the main SHAP vector.

The old vector decomposition approach is still available via the `legacy` parameter.

For testing purposes, legacy mode can be activated and de-activated for fitted inspectors by setting the `_legacy` attribute.

Orthogonalization of feature interaction vectors in the projection method can be enabled/disabled using the `orthogonalize` parameter.

### CODE REVIEW

File: src/facet/inspection/_inspection.py [hunk 1]
```diff
@@ -497,6 +554,9 @@ def feature_synergy_matrix(
         In the case of multi-target regression and non-binary classification, returns
         a list of data frames with one matrix per output.
 
+        :param absolute: if ``False``, return relative synergy as a percentage of
+            total feature importance;
+            if ``True``, return absolute synergy as a portion of feature importance
         :param symmetrical: if ``True``, return a symmetrical matrix quantifying
             mutual synergy; if ``False``, return an asymmetrical matrix quantifying
             unilateral redundancy of the features represented by rows with the
```
Comment (mgelsm) [inline src/facet/inspection/_inspection.py]:
```suggestion
            unilateral synergy of the features represented by rows with the
```

File: src/facet/inspection/_shap_projection.py [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+        self._ensure_fitted()
+        return self.association_.get_matrix(symmetrical=symmetrical, absolute=absolute)
+
+    def _fit(self, shap_calculator: ShapCalculator) -> None:
+        #
+        # basic definitions
+        #
+
+        shap_values: pd.DataFrame = shap_calculator.get_shap_values(consolidate="mean")
+        n_outputs: int = len(shap_calculator.output_names_)
```
Comment (mgelsm) [inline src/facet/inspection/_shap_projection.py]:
```suggestion
        # name of target columns (only one if binary classifier)
        n_outputs: int = len(shap_calculator.output_names_)
```

Comment (j-ittner) [reply src/facet/inspection/_shap_projection.py]: You mean, number of target columns? (not name)

Comment (mgelsm) [reply src/facet/inspection/_shap_projection.py]: yes :) 
I meant to specify the shap_calculator.output_names_ refers to the list of target names

Comment (j-ittner) [reply src/facet/inspection/_shap_projection.py]: Ah - not always which is why I call it outputs. This is more general than targets:

- In multi-class classification, there is one output column for every class although all of these relate to the same target.
- In multi-target regression, the outputs are indeed the targets.

**Review by jason-bentley — APPROVED** [review]
Awesome!

**Pull request merged**